### PR TITLE
Ahmad - Week2

### PR DESCRIPTION
**Pull request by amagsid**
HW for Databases week2

### CODE REVIEW

File: Week2/Homework/1.Keys.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    date_of_birth DATE,
+    h_index INTEGER,
+    gender ENUM('m', 'f'),
+    CONSTRAINT pk_authors PRIMARY KEY (author_no)
+  );`;
+
+  //mentor column
+  const addMentorColumn = `
+  ALTER TABLE authors
+  ADD mentor INTEGER,
```
Comment (unmeshvrije) [inline Week2/Homework/1.Keys.js:36]: I love ❤️  the indentation. Makes it much more readable! 🎉

File: Week2/Homework/2.Relationships.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    conference VARCHAR(150),
+    publish_date DATE,
+    PRIMARY KEY (paper_id)
+  );`;
+
+  //create an intermediary table representing the many-to-many relationship
+  //between the authors and research_papers tables
+  const createAuthorPapersTable = `
+  CREATE TABLE IF NOT EXISTS author_research_papers (
+    authorNO INT,
```
Comment (unmeshvrije) [inline Week2/Homework/2.Relationships.js:36]: perhaps `NOT NULL` ?

File: Week2/Homework/2.Relationships.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    paperID INT,
+    FOREIGN KEY (authorNO) REFERENCES authors(author_no),
+    FOREIGN KEY (paperID) REFERENCES research_papers(paper_id)
+  );`;
+
+  const mentorData = `
+  UPDATE authors
+    SET mentor = CASE
```
Comment (unmeshvrije) [inline Week2/Homework/2.Relationships.js:44]: This is a nice trick to update mentors!

File: Week2/Homework/2.Relationships.js (hunk 2 above)
Comment (unmeshvrije) [inline Week2/Homework/2.Relationships.js:36]: authorNO is an add name. I would rename it as `author_no` or `author_id`

File: Week2/Homework/4.AggrecateFunctions.js [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+  const researchPapersNumberOfAuthors = `
+  SELECT research_papers.paper_title AS research_papers, COUNT(authors.author_name) AS authors_number
+    FROM research_papers
+    INNER JOIN author_research_papers ON paperID = research_papers.paper_id
+    INNER JOIN authors ON author_research_papers.authorNO = authors.author_no
+    GROUP BY paper_title;
+  `;
+
+  const sumOfPapersWrittenByFemale = `
+    SELECT COUNT(research_papers.paper_title) AS papers_by_female_authors
```
Comment (unmeshvrije) [inline Week2/Homework/4.AggrecateFunctions.js:27]: What is two papers have the same title ?

File: Week2/Homework/4.AggrecateFunctions.js [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+    `;
+
+  const hIndexPerUniversity = `
+    SELECT university, AVG (h_index) AS average_h_index
+    FROM authors
+    GROUP BY university;
+    `;
+
+  const paperSumPerUniversity = `
+    SELECT university, COUNT (research_papers.paper_title) AS research_papers
```
Comment (unmeshvrije) [inline Week2/Homework/4.AggrecateFunctions.js:41]: Same as before: what if two papers have the same title, or multiple people wrote the same paper ?

Did you check if the query covers these cases ?

**Review by unmeshvrije — CHANGES_REQUESTED** [review]
Overall good work. Please check my comments

File: Week2/Homework/2.Relationships.js (hunk 3 above)
Comment (amagsid) [reply Week2/Homework/2.Relationships.js:44]: This one was my last resort after trying for a couple or so hours 😂

File: Week2/Homework/4.AggrecateFunctions.js (hunk 4 above)
Comment (amagsid) [reply Week2/Homework/4.AggrecateFunctions.js:27]: Good catch!

File: Week2/Homework/4.AggrecateFunctions.js (hunk 5 above)
Comment (amagsid) [reply Week2/Homework/4.AggrecateFunctions.js:41]: This one slipped by in the sea of errors I was trying to fix getting the HW for this week done 😂

**Comment by amagsid** [comment]
As always, thanks @unmeshvrije for taking the time. I fixed code according to your comments! 🚀

**Pull request closed without merging**